default.nix

```diff
@@ -39,92 +45,125 @@ let
+                  nixpkgs.lib.optionals rocksDbOptions.enableJemalloc [nixpkgs.jemalloc] ++
+                 [nixpkgs.git];
   });
-
-  # declares a build environment where C and C++ compilers are delivered by the llvm/clang project
-  # in this version build process should rely only on clang, without access to gcc
-  llvm = nixpkgs.llvmPackages_11;
-  env = llvm.stdenv;
-  llvmVersionString = "${nixpkgs.lib.getVersion env.cc.cc}";
+  rocksDbShellHook = if useCustomRocksDb
+                     then
+                       "export ROCKSDB_LIB_DIR=${customRocksdb}/lib; export ROCKSDB_STATIC=1"
+                     else "";
+
+  # newer versions of substrate support providing a version hash by means of an env variable, i.e. SUBSTRATE_CLI_GIT_COMMIT_HASH
+  gitFolder = builtins.path { path = ./.git; name = "git-folder"; };
+  gitCommitDrv = nixpkgs.runCommand "gitCommit" { nativeBuildInputs = [nixpkgs.git]; } ''
+    GIT_DIR=${gitFolder} git rev-parse --short HEAD > $out
+  '';
+  gitCommit = builtins.readFile gitCommitDrv;
+
+  modePath = if release then "release" else "debug";
+  pathToWasm = "target/" + modePath + "/wbuild/aleph-runtime/target/wasm32-unknown-unknown/" + modePath + "/aleph_runtime.wasm";
+  pathToCompactWasm = "target/" + modePath + "/wbuild/aleph-runtime/aleph_runtime.compact.wasm";
+
+  features =
+    builtins.concatLists
+      (builtins.attrValues
+        (builtins.mapAttrs
+          (package: features: builtins.map (feature: package + "/" + feature) features)
+          crates
+        )
+      );
+  enabledFeatures = nixpkgs.lib.concatStringsSep "," features;
+  featuresFlag = if enabledFeatures == "" then "" else "--features " + enabledFeatures;
+  packageFlags = builtins.map (crate: "--package " + crate) (builtins.attrNames crates);
 
   # allows to skip files listed by .gitignore
   # otherwise `nix-build` copies everything, including the target directory
-  gitignoreSrc = nixpkgs.fetchFromGitHub {
-    owner = "hercules-ci";
-    repo = "gitignore.nix";
-    rev = "5b9e0ff9d3b551234b4f3eb3983744fa354b17f1";
-    sha256 = "o/BdVjNwcB6jOmzZjOH703BesSkkS5O7ej3xhyO8hAY=";
+  inherit (versions.gitignore) gitignoreFilter;
+  # we need to include the .git directory, since Substrate build scripts use git to retrieve commit hash of HEAD
+  gitFilter = src:
+    let
+      srcIgnored = gitignoreFilter src;
+    in
+      path: type:
+        builtins.baseNameOf path == ".git" || srcIgnored path type;
+  src = nixpkgs.lib.cleanSourceWith {
+    src = ./.;
+    filter = gitFilter ./.;
+    name = "aleph-source";
   };
-  inherit (import gitignoreSrc { inherit (nixpkgs) lib; }) gitignoreSource;
 in
-with nixpkgs; env.mkDerivation rec {
-  name = "aleph-node";
-  src = gitignoreSource ./.;
-
-  buildInputs = [
-    rustToolchain
-    llvm.clang
-    openssl.dev
-    protobuf
-    customRocksdb
-    pkg-config
-    cacert
+with nixpkgs; naersk.buildPackage rec {
+  inherit name release src singleStep;
+  doCheck = runTests;
```

Comment:
Can we please have syntax with `@` here to name this script arguments via `args` aliast